src/pymatgen/core/composition.py

```diff
@@ -639,22 +640,52 @@ def from_dict(cls, dct: dict) -> Self:
         return cls(dct)
 
     @classmethod
```

Comment:
I think adding a few init examples for this method would be useful. Similar to the one at the init method of Composition.